nodes/ElseNode.java

```diff
@@ -45,7 +67,19 @@ public String convertToPython() {
 
     @Override
     public boolean validateTree() {
-        return false;
+
+        // validate if else node is preceded by an if
+        if (!this.getHasIfParent()) {
+            throw new SemanticException("'else' without 'if'", fileInfo.getFilename(), fileInfo.getLineNum());
+            return false;
```

Comment:
Since you're throwing an error, this won't be reached. I'd just recommend removing this?